### ISSUE TITLE
Calendar contents reset in TB 128, and synckey prevents retrieval

### ISSUE THREAD

**Issue by llvilanova** [jobisoft/EAS-4-TbSync#274]
## Your environment

TbSync version: 4.11
EAS-4-TbSync version: 4.12
Thunderbird version: 128.1.1esr

## Expected behavior
Calendar in my office365 account gets populated with all entries created in the past.

## Actual behavior
Only newly created calendar entries (since upgrade to TB 128 and respective TbSync plugin updates) are shown in TB's calendar. Old entries are not retrieved even after I tell TbSync that "synchronization period" is "all".

Using the debugger to force "SyncKey" to "0" in sync.requestRemoteChanges (sync.js:404) results in Sync error code 4 from the server.

## Steps to reproduce
In my case, just upgrading to TB 128 and respective TbSync plugin updates.

**Comment by jobisoft** [comment]
Are you using the Flatpack version of Thunderbird 128? Or Snap?

**Comment by jobisoft** [comment]
Since I have not changed the version for TB115 and this is reported by users of the OLD version for TB115 and the NEW version for TB128, Microsoft changed something. I fear this is out of my control.

**Comment by ottomano** [comment]
> Since I have not changed the version for TB115 and this is reported by users of the OLD version for TB115 and the NEW version for TB128, Microsoft changed something. I fear this is out of my control.

Please keep us posted on this. I just reinstalled TB 128 and now I can't retrieve all my calendar entries. 😕

**Comment by jobisoft** [comment]
I do not think this is related to Thunderbird 128. Reports indicate that Thunderbird 115 is affected as well. It appears only certain Microsoft services are affected.

**Comment by jobisoft** [comment]
If you enable debug logging, you will find such an entry on all fresh synced (disabled and then re-enabled) calendars:

```
<?xml version="1.0"?>
<GetItemEstimate xmlns='GetItemEstimate'>
<Response>
<Status>1</Status>
<Collection>
<CollectionId>14</CollectionId>
<Estimate>0</Estimate>
</Collection>
</Response>
</GetItemEstimate>
```

The server sends that he has no data for calendars. It does send the correct estimate for tasks.

Microsoft messed up their server.

**Comment by damien-git** [comment]
I am not sure what the estimate is used for in EAS-4-TbSync, but is it possible you would get data if you simply ignore it ? It is just an "estimate" after all...

**Comment by jobisoft** [comment]
That estimate reference was just a hint. I am requesting the full changes after the estimate. But that request comes back empty.

**Comment by damien-git** [comment]
I think ActiveSync can be used by Outlook clients if configured to, to access 365 calendars. There was an [issue](https://www.bleepingcomputer.com/news/microsoft/microsoft-fixes-outlook-clients-not-syncing-over-exchange-activesync/) in February affecting them. So shouldn't this change affect them as well ?

**Comment by jobisoft** [comment]
If they really do use EAS, they could be affected as well. But I assume they use EWS instead. But this is all just guessing. Fact is I follow the spec and it worked for multiple years and now it no longer works for MS.

**Comment by llvilanova** [comment]
> Are you using the Flatpack version of Thunderbird 128? Or Snap?

I'm using the version packaged in Debian.

**Comment by wiraki** [comment]
> I do not think this is related to Thunderbird 128. Reports indicate that Thunderbird 115 is affected as well. It appears only certain Microsoft services are affected.

Can confirm on 115 .deb version as well.

Thanks for all the info. What a bummer :(
Any suggestions where to look for updates/fixes on this?

**Comment by schillic** [comment]
It just started working again for me. Hooray!

**Comment by bubble75** [comment]
One more observation: If event arrives while Thunderbird is running with ActiveSync account is enabled - it is shown in calendar. Once I disable and reenable account or restart Thunderbird - it is no longer shown.

**Comment by bubble75** [comment]
> One more observation: If event arrives while Thunderbird is running with ActiveSync account is enabled - it is shown in calendar. Once I disable and reenable account or restart Thunderbird - it is no longer shown.

Just started to work again.

**Comment by ottomano** [comment]
Still not working for me on Office 365 calendar, but now I have big hopes after it started working for others. (fingers crossed)

**Comment by schor123** [comment]
Can confirm it started working again. I removed all calendars, uninstalled tbsync; re-started Thunderbird, re-installed tbsync + exchange provider and there we go: all calendars/tasks are back.

**Comment by ottomano** [comment]
> Tried all that: removing calendars, uninstalling TbSync and EAS-4-Tbsync, restarting Thunderbird, re-installing the extensions and re-adding Office365 calendars - still nothing.

Same issue here, unfortunately.

**Comment by faz17** [comment]
> Tried all that: removing calendars, uninstalling TbSync and EAS-4-Tbsync, restarting Thunderbird, re-installing the extensions and re-adding Office365 calendars - still nothing.

Same here

**Comment by LamyLab** [comment]
Same as geg-ck.

**Comment by jobisoft** [comment]
It may be helpful to note which platform. So instead of **"same here"**, post something like **"outlook.com: OK"** or **"office365: NOK"**

**Comment by NEP1192** [comment]
Office365, not working

**Comment by geg-ck** [comment]
Microsoft365 Exchange Online - OK here.

**Comment by eamathw** [comment]
outlook.office365.com working
TbSync v4.8
Provider for Exchange ActiveSync v4.8

**Comment by jobisoft** [comment]
> outlook.office365.com working TbSync v4.8 Provider for Exchange ActiveSync v4.8

That is even better!

**Comment by presianbg** [comment]
Still not working:
- EU
- Microsoft 365 (Exchange Online)
- TbSync v4.11
- Provider for Exchange ActiveSync 4.12

**Comment by mlassnig** [comment]
Not working:

- TB 128.2.0esr / Debian
- Provider for Exchange ActiveSync 4.12
- TbSync 4.11
- outlook.office365.com

Repeated installs/refreshs... same effect that only new events get loaded in.

**Comment by ottomano** [comment]
- TB 128.2.0 esr / Windows 10
- outlook.office365.com

Deleted both TBsync and EAS, rebooted PC, reinstalled both plugins. unfortunately it's not working.

@jobisoft I can provide you debug log via DM if needed

**Comment by jobisoft** [comment]
> * TB 128.2.0 esr / Windows 10
> 
>     * outlook.office365.com
> 
> 
> Deleted both TBsync and EAS, rebooted PC, reinstalled both plugins. unfortunately it's not working.
> 
> @jobisoft I can provide you debug log via DM if needed

You only have to disable and re-enable the TbSync account to see if entries are coming back. No need to uninstall anything. Check your debug log for the estimate entry as shown here:
https://github.com/jobisoft/EAS-4-TbSync/issues/274#issuecomment-2338395919

If the estimate for all your calendars is 0, your server is still not reporting any entries.

**Comment by ottomano** [comment]
> > * TB 128.2.0 esr / Windows 10
> >   
> >   * outlook.office365.com
> > 
> > Deleted both TBsync and EAS, rebooted PC, reinstalled both plugins. unfortunately it's not working.
> > @jobisoft I can provide you debug log via DM if needed
> 
> You only have to disable and re-enable the TbSync account to see if entries are coming back. No need to uninstall anything. Check your debug log for the estimate entry as shown here: [#274 (comment)](https://github.com/jobisoft/EAS-4-TbSync/issues/274#issuecomment-2338395919)
> 
> If the estimate for all your calendars is 0, your server is still not reporting any entries.

Yes, I did it but with same result.

Here the XML response

<?xml version="1.0"?>
<GetItemEstimate xmlns='GetItemEstimate'>
<Collections>
<Collection>
<SyncKey xmlns='AirSync'>11187586</SyncKey>
<CollectionId>6</CollectionId>
<Options xmlns='AirSync'>
<FilterType xmlns='AirSync'>7</FilterType>
<Class xmlns='AirSync'>Calendar</Class>
</Options>
</Collection>
</Collections>
</GetItemEstimate>

**Comment by jobisoft** [comment]
> > > * TB 128.2.0 esr / Windows 10
> > >   
> > >   * outlook.office365.com
> > > 
> > > Deleted both TBsync and EAS, rebooted PC, reinstalled both plugins. unfortunately it's not working.
> > > @jobisoft I can provide you debug log via DM if needed
> > 
> > 
> > You only have to disable and re-enable the TbSync account to see if entries are coming back. No need to uninstall anything. Check your debug log for the estimate entry as shown here: [#274 (comment)](https://github.com/jobisoft/EAS-4-TbSync/issues/274#issuecomment-2338395919)
> > If the estimate for all your calendars is 0, your server is still not reporting any entries.
> 
> Yes, I did it but with same result.
> 
> Here the XML response
> 
> ```
> 
> <?xml version="1.0"?>
> <GetItemEstimate xmlns='GetItemEstimate'>
> <Collections>
> <Collection>
> <SyncKey xmlns='AirSync'>11187586</SyncKey>
> <CollectionId>6</CollectionId>
> <Options xmlns='AirSync'>
> <FilterType xmlns='AirSync'>7</FilterType>
> <Class xmlns='AirSync'>Calendar</Class>
> </Options>
> </Collection>
> </Collections>
> </GetItemEstimate>
> ```

That is the request.

**Comment by ottomano** [comment]
> > > > * TB 128.2.0 esr / Windows 10
> > > >   
> > > >   * outlook.office365.com
> > > > 
> > > > Deleted both TBsync and EAS, rebooted PC, reinstalled both plugins. unfortunately it's not working.
> > > > @jobisoft I can provide you debug log via DM if needed
> > > 
> > > 
> > > You only have to disable and re-enable the TbSync account to see if entries are coming back. No need to uninstall anything. Check your debug log for the estimate entry as shown here: [#274 (comment)](https://github.com/jobisoft/EAS-4-TbSync/issues/274#issuecomment-2338395919)
> > > If the estimate for all your calendars is 0, your server is still not reporting any entries.
> > 
> > 
> > Yes, I did it but with same result.
> > Here the XML response
> > ```
> > 
> > <?xml version="1.0"?>
> > <GetItemEstimate xmlns='GetItemEstimate'>
> > <Collections>
> > <Collection>
> > <SyncKey xmlns='AirSync'>11187586</SyncKey>
> > <CollectionId>6</CollectionId>
> > <Options xmlns='AirSync'>
> > <FilterType xmlns='AirSync'>7</FilterType>
> > <Class xmlns='AirSync'>Calendar</Class>
> > </Options>
> > </Collection>
> > </Collections>
> > </GetItemEstimate>
> > ```
> 
> That is the request.

Sorry, my fault.

```
<?xml version="1.0"?>
<GetItemEstimate xmlns='GetItemEstimate'>
<Response>
<Status>1</Status>
<Collection>
<CollectionId>6</CollectionId>
<Estimate>0</Estimate>
</Collection>
</Response>
</GetItemEstimate>

```

**Comment by jobisoft** [comment]
Yep, your server does not report any data for you. I guess you have to keep waiting. Maybe Microsoft is rolling out the fix in waves and it does not hit all servers at the same time.

If that estimate comes back non-zero and you *still* do not get data, I can start looking at it again. Until then, you have to be patient. Disable and re-enable your TbSync account once a day and see if a fresh sync brings back any data and/or it reports a non-zero estimate.

**Comment by bermeitinger-b** [comment]
It started working for me. So patience is the only way to go :)

Thanks!

**Comment by faz17** [comment]
Tbsync 4.11
Provider Activesync 4.12
thunderbird 128.2.0 esr
windows 11 
outlook.office365.com
EU
still not working. Here is the log
```
<?xml version="1.0"?>
<GetItemEstimate xmlns='GetItemEstimate'>
<Collections>
<Collection>
<SyncKey xmlns='AirSync'>2047807048</SyncKey>
<CollectionId>9</CollectionId>
<Options xmlns='AirSync'>
<FilterType xmlns='AirSync'>7</FilterType>
<Class xmlns='AirSync'>Calendar</Class>
</Options>
</Collection>
</Collections>
</GetItemEstimate>
```

**Comment by presianbg** [comment]
> Still not working:
> 
> * EU
> * Microsoft 365 (Exchange Online) (outlook.office365.com)
> * TB 128.2.0esr
> * TbSync v4.11
> * Provider for Exchange ActiveSync 4.12

It started working... Microsoft being Microsoft :)

**Comment by ottomano** [comment]
> > > > > * TB 128.2.0 esr / Windows 10
> > > > >   
> > > > >   * outlook.office365.com
> > > > > 
> > > > > Deleted both TBsync and EAS, rebooted PC, reinstalled both plugins. unfortunately it's not working.
> > > > > @jobisoft I can provide you debug log via DM if needed
> > > > 
> > > > 
> > > > You only have to disable and re-enable the TbSync account to see if entries are coming back. No need to uninstall anything. Check your debug log for the estimate entry as shown here: [#274 (comment)](https://github.com/jobisoft/EAS-4-TbSync/issues/274#issuecomment-2338395919)
> > > > If the estimate for all your calendars is 0, your server is still not reporting any entries.
> > > 
> > > 
> > > Yes, I did it but with same result.
> > > Here the XML response
> > > ```
> > > 
> > > <?xml version="1.0"?>
> > > <GetItemEstimate xmlns='GetItemEstimate'>
> > > <Collections>
> > > <Collection>
> > > <SyncKey xmlns='AirSync'>11187586</SyncKey>
> > > <CollectionId>6</CollectionId>
> > > <Options xmlns='AirSync'>
> > > <FilterType xmlns='AirSync'>7</FilterType>
> > > <Class xmlns='AirSync'>Calendar</Class>
> > > </Options>
> > > </Collection>
> > > </Collections>
> > > </GetItemEstimate>
> > > ```
> > 
> > 
> > That is the request.
> 
> Sorry, my fault.
> 
> ```
> <?xml version="1.0"?>
> <GetItemEstimate xmlns='GetItemEstimate'>
> <Response>
> <Status>1</Status>
> <Collection>
> <CollectionId>6</CollectionId>
> <Estimate>0</Estimate>
> </Collection>
> </Response>
> </GetItemEstimate>
> ```

it started "working" again. But it does not download all entries once. Seems it is slowly downloading few entries each hour (so, now my calendar on TB is roughly at 50% of the real calendar on Outlook).

**Comment by SirFritz** [comment]
This never started working again for me.  Instead now on two different computers it locks up Thunderbird when trying to sync calenders.

**Comment by damien-git** [comment]
It finally started working for me too (O365, TB128). I used a read-only ics link while I was waiting.

**Comment by faz17** [comment]
And finally it's working again also for me!
